test(agent-state): 4 integration tests in AgentStateDetection not updated after PR #4999 introduced exited state

## Summary

4 integration tests in `electron/services/__tests__/AgentStateDetection.integration.test.ts` were not updated when PR #4999 (commit `c1d85ba40`) changed the `AgentStateMachine` exit event behavior from `"completed"` → `"exited"`. They will never pass in their current state.

## Failing Tests

```
× should handle state transitions for agent terminals              2610ms
  Error: Timeout — waited 2000ms for state "completed", never arrived

× should preserve completed state across project switches           813ms
  AssertionError: expected 'exited' to be 'completed'
  at AgentStateDetection.integration.test.ts:463

× should preserve failed state across project switches             906ms
  AssertionError: expected 'exited' to be 'completed'
  at AgentStateDetection.integration.test.ts:487

× should maintain accurate agent state for background terminals    1010ms
  AssertionError: expected 'exited' to be 'completed'
  at AgentStateDetection.integration.test.ts:630
```

Reproduced on: Ubuntu 24.04.4 LTS, kernel 6.17.0-20-generic, x86_64 (ASUS VivoBook X415EA)

## Root Cause

`AgentStateMachine.ts` was changed in commit `c1d85ba40`:

```diff
case "exit":
-  if (current === "working" || current === "waiting" || current === "completed") {
-    return "completed";
+  if (current !== "idle" && current !== "exited") {
+    return "exited";
  }
```

`AgentStateMachine.test.ts` was correctly updated in that commit. `AgentStateDetection.integration.test.ts` does not appear in the commit's changeset — it was missed.

The unit tests confirm the new behavior is intentional:
```typescript
// AgentStateMachine.test.ts:157-171 (passing)
expect(nextAgentState("working", { type: "exit", code: 0 })).toBe("exited");
expect(nextAgentState("working", { type: "exit", code: 1 })).toBe("exited");
```

## Lines to Fix

| File | Line(s) | Change |
|------|---------|--------|
| `AgentStateDetection.integration.test.ts` | 193 | `waitForAgentStateChange(..., "completed")` → `"exited"` |
| `AgentStateDetection.integration.test.ts` | 197 | `.toBe("completed")` → `.toBe("exited")` |
| `AgentStateDetection.integration.test.ts` | 463, 468 | `.toBe("completed")` → `.toBe("exited")` |
| `AgentStateDetection.integration.test.ts` | 487, 492 | `.toBe("completed")` → `.toBe("exited")` |
| `AgentStateDetection.integration.test.ts` | 630 | `.toBe("completed")` → `.toBe("exited")` |
| `AgentStateDetection.integration.test.ts` | 471 | Test name: rename `"failed state"` → `"exited state"` (references removed state) |

## Reporter

- **OS:** Ubuntu 24.04.4 LTS (Noble Numbat)
- **Kernel:** 6.17.0-20-generic, x86_64
- **Machine:** ASUS VivoBook X415EA

--- a/electron/services/__tests__/AgentStateDetection.integration.test.ts
+++ b/electron/services/__tests__/AgentStateDetection.integration.test.ts
@@ -186,15 +186,15 @@ describe.skipIf(shouldSkip)("Agent State Detection Integration", () => {
       const id = await spawnShellTerminal(manager, { type: "claude" });
       await sleep(500);
 
-      // First transition to working, then to completed (exit from idle doesn't work)
+      // First transition to working, then to exited (exit from idle doesn't work)
       manager.transitionState(id, { type: "busy" }, "activity", 1.0);
       await sleep(100);
 
-      const statePromise = waitForAgentStateChange(manager, id, 2000, "completed");
+      const statePromise = waitForAgentStateChange(manager, id, 2000, "exited");
       manager.transitionState(id, { type: "exit", code: 0 }, "activity", 1.0);
 
       const stateChange = await statePromise;
-      expect(stateChange.state).toBe("completed");
+      expect(stateChange.state).toBe("exited");
     }, 10000);
 
     it("should handle state transitions for different agent types", async () => {
@@ -447,7 +447,7 @@ describe.skipIf(shouldSkip)("Agent State Detection Integration", () => {
       expect(terminalAfter?.lastStateChange).toBe(preservedLastStateChange);
     }, 10000);
 
-    it("should preserve completed state across project switches", async () => {
+    it("should preserve exited state across project switches", async () => {
       const projectId = "test-project-456";
       const id = await spawnShellTerminal(manager, { type: "claude" });
       await sleep(500);
@@ -460,15 +460,15 @@ describe.skipIf(shouldSkip)("Agent State Detection Integration", () => {
 
       const terminal = manager.getTerminal(id);
       expect(terminal).toBeDefined();
-      expect(terminal?.agentState).toBe("completed");
+      expect(terminal?.agentState).toBe("exited");
 
       await sleep(500);
 
       const terminalAfter = manager.getTerminal(id);
-      expect(terminalAfter?.agentState).toBe("completed");
+      expect(terminalAfter?.agentState).toBe("exited");
     }, 10000);
 
-    it("should preserve failed state across project switches", async () => {
+    it("should preserve exited state across project switches (crash exit)", async () => {
       const projectId = "test-project-789";
       const id = await spawnShellTerminal(manager, { type: "gemini" });
       await sleep(500);
@@ -484,12 +484,12 @@ describe.skipIf(shouldSkip)("Agent State Detection Integration", () => {
 
       const terminal = manager.getTerminal(id);
       expect(terminal).toBeDefined();
-      expect(terminal?.agentState).toBe("completed");
+      expect(terminal?.agentState).toBe("exited");
 
       await sleep(500);
 
       const terminalAfter = manager.getTerminal(id);
-      expect(terminalAfter?.agentState).toBe("completed");
+      expect(terminalAfter?.agentState).toBe("exited");
     }, 10000);
   });
 
@@ -627,7 +627,7 @@ describe.skipIf(shouldSkip)("Agent State Detection Integration", () => {
       await sleep(100);
 
       const stateAfterExit = manager.getTerminal(terminalB);
-      expect(stateAfterExit?.agentState).toBe("completed");
+      expect(stateAfterExit?.agentState).toBe("exited");
     }, 10000);
   });
 });

--- a/electron/services/__tests__/AgentStateDetection.integration.test.ts
+++ b/electron/services/__tests__/AgentStateDetection.integration.test.ts
@@ -455,7 +455,10 @@ describe.skipIf(shouldSkip)("Agent State Detection Integration", () => {
       manager.onProjectSwitch(projectId);
       await sleep(100);
 
-      manager.transitionState(id, { type: "exit", code: 0 }, "activity", 1.0);
+      // First transition to working, then trigger a clean exit (exit events are no-ops from idle)
+      manager.transitionState(id, { type: "busy" }, "activity", 1.0);
+      await sleep(100);
+      manager.transitionState(id, { type: "exit", code: 0 }, "exit", 1.0);
       await sleep(200);
 
       const terminal = manager.getTerminal(id);

--- a/electron/services/__tests__/AgentStateDetection.integration.test.ts
+++ b/electron/services/__tests__/AgentStateDetection.integration.test.ts
@@ -257,7 +257,7 @@ describe.skipIf(shouldSkip)("Agent State Detection Integration", () => {
       const id = await spawnShellTerminal(manager, { type: "claude" });
       await sleep(500);
 
-      // states to transition to: "working", "waiting", "working", "completed"
+      // states to transition to: "working", "waiting", "working", "exited"
       // events: { type: "busy" }, { type: "prompt" }, { type: "busy" }, { type: "exit", code: 0 }
 
       const events = [
@@ -274,7 +274,7 @@ describe.skipIf(shouldSkip)("Agent State Detection Integration", () => {
 
       const terminal = manager.getTerminal(id);
       expect(terminal).toBeDefined();
-      // Final state should be completed, but we just check if it's defined
+      // Final state should be exited, but we just check it's defined to avoid timing sensitivity
       expect(terminal?.agentState).toBeDefined();
     }, 10000);
 

